test/application/certificate.test.js

```diff
@@ -0,0 +1,197 @@
+      });
+
+      it('should set revocationKey of the certificate object', () => {
+        expect(certificate.revocationKey).toEqual(null);
+      });
+
+      it('should set sealImage of the certificate object', () => {
+        expect(certificate.sealImage).toEqual(MainnetV2Certificate.badge.issuer.image);
+      });
+
+      it('should set signature of the certificate object', () => {
+        expect(certificate.signature).toEqual(null);
+      });
+
+      it('should set 1 signatureImage to the certificate object', () => {
+        expect(certificate.signatureImage.length).toEqual(1);
+      });
+
+      it('should set subtitle to the certificate object', () => {
+        expect(certificate.subtitle).toEqual(MainnetV2Certificate.badge.subtitle);
+      });
+
+      it('should set name to the certificate object', () => {
+        expect(certificate.name).toEqual(MainnetV2Certificate.badge.name);
+      });
+
+      it('should set transactionId to the certificate object', () => {
+        expect(certificate.transactionId).toEqual(MainnetV2Certificate.signature.anchors[0].sourceId);
+      });
+
+      it('should set rawTransactionLink to the certificate object', () => {
+        const rawTransactionLinkAssertion = `https://blockchain.info/rawtx/${MainnetV2Certificate.signature.anchors[0].sourceId}`;
+        expect(certificate.rawTransactionLink).toEqual(rawTransactionLinkAssertion);
+      });
+
+      it('should set transactionLink to the certificate object', () => {
+        const transactionLinkAssertion = `https://blockchain.info/tx/${MainnetV2Certificate.signature.anchors[0].sourceId}`;
+        expect(certificate.transactionLink).toEqual(transactionLinkAssertion);
+      });
+
+      it('should set verificationSteps to the certificate object', () => {
+        expect(certificate.verificationSteps).toEqual([]);
+      });
+
+      it('should set isFormatValid to the certificate object', () => {
+        expect(certificate.isFormatValid).toEqual(true);
+      });
+
+      it('should set documentToVerify accordingly', () => {
+        const assertion = Object.assign({}, MainnetV2Certificate);
+        delete assertion['signature'];
+        expect(certificate.documentToVerify).toEqual(assertion);
+      });
+    });
+
+    describe('given it is called with invalid certificate data', () => {
+      it('should return an error', () => {
+        expect(() => {
+          /* eslint no-unused-vars: "off" */
+          let certificate = new Certificate('invalid-certificate-data');
+        }).toThrowError('This is not a valid certificate');
+      });
+    });
+
+    describe('given it is called with no certificate data', () => {
+      it('should throw an error', () => {
+        expect(() => {
+          /* eslint no-unused-vars: "off" */
+          let certificate = new Certificate();
+        }).toThrowError('This is not a valid certificate');
+      });
+    });
+  });
```

Comment:
I think we should also run these tests against a v1 version.

Reply:
You could/should do it in a separate file though, so that we don't bloat this one.